src/connectors/userService.ts

```diff
@@ -600,7 +600,49 @@ export class UserService extends BaseService {
     viewerId?: string | null
     exclude?: GQLSearchExclude
   }) => {
-    return { nodes: [], totalCount: 0 }
+    const displayName = key
+    const userName =
+      key.startsWith('@') || key.startsWith('＠') ? key.slice(1) : key
+
+    if (!userName) {
+      return { nodes: [], totalCount: 0 }
+    }
+
+    const baseQuery = this.knex('search_index.user')
+      .select('id')
+      .whereLike('display_name', `%${displayName}%`)
+      .orWhereLike('user_name', `%${userName}%`)
+      .orderByRaw('display_name = ? DESC', [displayName])
```

Comment:
here need to check if search key startsWith('@'), then search on user_name only, no matching on `display_name`